### ISSUE TITLE
Improvements, pt. 2: 

### ISSUE THREAD

**Issue by willson556** [GetiPlayerAutomator/get-iplayer-automator#69]
_From [jim.sinf...@gmail.com](https://code.google.com/u/114776053600719259070/) on February 16, 2012 05:24:44_

Wasn't sure whether to sub these as separate Issue #'s, but thought it was easier to deal with them in one place... 

(1a) Download next queued item, while converting previous item in queue. 

(1b) (...or even better) Simultaneous downloads. Perhaps up to three items at once as an option in Preferences. 
For those of us on good connections, surely possible? The official iPlayer Desktop app automatically does this AFAIR (since GIA, I haven't had to use it in a while, lol!). 

(2) Failed to convert ".partial" downloads: allow users to manually hit a button to re-attempt to convert file. 
Often the .flv file downloads in full, but then mysteriously fails to convert to a .mp4 file. Other converting solutions I've tried take ages, as the container has to be converted in real time to a .mp4 file (are there other good/quicker solutions out there?). If the user leaves the .partial file in the same download location as they stipulated in Prefs as the Download Path, having a "Manual Convert" button in the main app window, would help greatly as a first solution for users at least. 

(3) Download Queue: manually reorder items. 
While we can use the column headers "PID"/"Status"/"Show Name - Episode Name" to do a reorder of sorts, there is no way to manually put an item to the top/reorder the queue in an way. We really need to be able to manually drag them to our preferred order, especially when we need something that's about to be removed from iPlayer very soon to download _first_ before other items. Annoyingly, currently we have to delete the other items > then add the one(s) needed first > then re-search to add the other items again. 

(4) Some kind of guide to server speed, if possible. 
Often a group of files that may take 20 mins on one day, suddenly takes 120 mins the next, even though my connection is working fine at expected speed (I check). So I'm presuming it's a file-server slowdown at the BBC/ITV end, but hard to really know. It would be great to somehow have an idea if the server is causing the speed blockage, rather than our internet connections. An in-app MB/s speed guide (perhaps where &#37; downloaded info is) may also help. 

(5) Folders that always exist on downloaded files: add option to auto-remove/not have in the first place. 
When I download a queue of programmes, I always get separate folders with each programme in its own folder. Most of the time I don't need said folder, hence dragging file out of folder, then deleting the folder. How about a Prefs option to leave the file directly in the Download Path location we stipulate in Prefs and delete the folder (I presume I'm right in thinking the separate folder is needed in the tagging/conversion process — can the tagging/conversion not be done directly in the Download Path location folder itself?). 
eg. I download a series (same for one-off) programme like "The Singing Detective: 2. Heat": I get series folder "The Singing Detective", with file "The Singing Detective - 2. Heat ((flashhd)).mp4" in it — unnecessary separate folder. Same with a film "Heat": file "Heat - Heat ((flashvhigh)).mp4", folder "Heat" — again folder unnecessary. 

(6a) Download History window options: "Select all" (cmd+A shortcut). 
Going to File>Edit Download History, cmd+A doesn't work here to select all items at once for deletion, rather than doing one at a time. Yes, I know we can also go to ~Library/Application Support/Get iPlayer Automator and delete the whole "downloads_history" file, but this would be much more convenient & quicker when we need to delete the whole lot. 

(6b) Download History window options: issues with Save button + the Save button needs keyboard shortcut abilities enabled. 
Firstly, even after an item has been removed from the DH list and Save button pressed, we have to refresh the cache before we can re-download it – why? 
Secondly, following the removal of an item from the DH list, if the Save button is _not_ then pressed, we are still able to shut the DH window, and the item will still show as visibly disappeared from the DH list, even though the item has not been properly removed from it. If we reopen the DH window, even though the DH list doesn't show the item on it, the app will still not allow it to be downloaded again. This is a somewhat confusing method for users. A pop-up dialog window with "Save" or "Cancel" options on it should appear beforehand if something has been deleted from the DH list, _before_ we are able to close the DH window at all. 
Thirdly, currently we have to point and click the Save/Cancel buttons to action them. We should be able to remove an item from the DH list using the backspace key, and then hit the return key to Save (or cmd+C to Cancel), then cmd+W to shut the DH window. If cmd+W (or clicking the red button) is attempted to be used to close the DH window _without first_ saving or cancelling —only if an item has actually been removed from the list first— then have the "Save"/"Cancel" options pop-up dialog window appear _with_ Save highlighted in blue, able to be confirmed by hitting the return key (or cmd+C to Cancel) and _then_ shutting the DH window. 
This is how most other apps work, so it's strange not having them on GIA. Might be worth testing all the current actions out, whilst fixing these. 

(6c) Download History window options: open and delete items whilst Download Queue still downloading. 
Having to often re-download items that fail (mainly because manually converting them outside GIA is more of a pain; see point (2) above), at the moment we have to wait until our current queue has been worked through (or stop it early), and then go to File>Edit Download History, and delete the item(s) before refreshing the cache, and then downloading item again. 

(6d) Download History window options: do NOT store items in a Download History folder, period. 
Can we have an option to not store any download history? For those of us that don't need it. 
I presume Download History folder is for Series-Link/PVR functions, but for those that don't use this, we often don't want to keep our download history. (perhaps do some sort of grey-out of the tick box in Prefs, or simply add a note next to the option in Prefs saying "Do not select if using Series-Link/PVR function".) 

(7a) Series-Link/PVR: in the PVR Manager, add the ep# of the LAST episode that was chosen (or was downloaded) to the "Series-Link:" queue's title. 
eg. In the PVR search results, if I hit "Doctor Who: Series 2 - 6. The Idiot's Lantern", then in the Series-Link queue please make it show "Doctor Who: Series 2 - 6. The Idiot's Lantern" (rather than just "Doctor Who: Series 2" it currently does). Then at least we know what the next ep# will be. And of course, it'll have to update to the next ep when ep7 is downloaded, etc. 

(7b) Series-Link/PVR: add the PVR manager as a (hidden) panel in the main app window, rather than separate floating window. 
Under the Search Results and Download Qu...

_Original issue: http://code.google.com/p/get-iplayer-automator/issues/detail?id=69_
